ivy/functional/frontends/tensorflow/linalg.py

```diff
@@ -253,3 +253,22 @@ def einsum(equation, *inputs, **kwargs):
 @to_ivy_arrays_and_back
 def adjoint(matrix, name=None):
     return ivy.adjoint(matrix)
+
+
+@with_supported_dtypes(
+    {
+        "2.9.0 and below": (
+            "float16",
```

Comment:
This "float16", is not supported by `linalg` in any framework as far as I know.
Kindly make sure you are using this decorator correctly :slightly_smiling_face: 

Also, as far as the casting error goes, can you please dive more deep and let me know why and where that occurs? :slightly_smiling_face: